### PR TITLE
document the @asynccontextmanager do-not-yield exemption

### PR DESCRIPTION
**Pull request by graingert**

### CODE REVIEW

File: docs/cancellation.rst [hunk 1]
```diff
@@ -153,3 +153,19 @@ The problem with this code is that it violates structural concurrency: what happ
 task raises an exception? The host task would be cancelled as a result, but the host task might be
 long gone by the time that happens. Even if it weren't, any enclosing ``try...except`` in the
 generator would not be triggered. In other words: it's a Really Bad Idea to do this!
+
+Async generators decorated with ``@asynccontextmanager`` to serve as the template for an async context
```
Comment (graingert) [inline docs/cancellation.rst:157]: lifted from trio https://trio.readthedocs.io/en/stable/reference-core.html#cancel-scopes-and-nurseries

**Comment by graingert** [comment]
related to #282

**Comment by frederikaalund** [comment]
Glad to see this as part of the documentation. :+1: 

I suggest that we add an "accepted use" example as well. Something like:

> You may do this:
>
>     @asynccontextmanager
>     async def some_cm():
>         async with create_task_group() as tg:
>             tg.start_soon(foo)
>             yield
>
> and use it like this:
>
>     async with some_cm():
>          ...
>
> but do **NOT** do this:
>
>     async def some_generator():
>         async with some_cm():
>             yield

**Comment by agronholm** [comment]
I would likely rewrite that explanation. I don't think it really explains well enough _why_ async context managers are okay in this regard (and of course, when using `__aenter__()` and `__aexit__()` explicitly, all bets are off with those too).

**Comment by agronholm** [comment]
I have opened an alternative PR as #294.

**Comment by agronholm** [comment]
Closing in favor of the merged PR #294.

**Pull request closed without merging**